### DISCUSSION TITLE
Creating folder sharing links to specific people

### DISCUSSION THREAD

**Discussion by GaddeChaitanya** [pnp/pnpcore#913] [Q&A] [answered]
Hi,

For sharing a folder using a link I have 4 options, among these I want to use 4th option which is "Specific People". 

Can you suggest me a method that performs this operation. I looked through the documentation but none of them are performing the way I intended.

**Note:**  Expectations are once link is shared then user should be placed under "Links giving access" instead of "Direct access" section.

![SharePoint Manage Access](https://user-images.githubusercontent.com/34321716/176902368-e897b06d-52ae-4815-82c9-573cacfaa3c1.jpg)

**1. Link Setting Options:**
![SharePoint Likn Share Options](https://user-images.githubusercontent.com/34321716/176901067-8f69bd0e-6633-4e40-bf8b-4deba5190ea4.jpg)

**2. Send Link:**
![SharePoint Send Link](https://user-images.githubusercontent.com/34321716/176902000-28a3d35c-7990-44b7-bd74-c87c37d335b5.jpg)

**Comment by MathijsVerbeeck** [comment, marked as answer]
Hi @GaddeChaitanya 

The following should do the trick: [https://pnp.github.io/pnpcore/using-the-sdk/sharing-intro.html#share-content-with-certain-users](https://pnp.github.io/pnpcore/using-the-sdk/sharing-intro.html#share-content-with-certain-users).

If you have any more questions on how to implement this, don't hesitate to ask them.

**Comment by GaddeChaitanya** [comment]
Thank you for sharing the documentation link. I was able to use "CreateUserSharingLink" method to give share links with specific user.
